main_test.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"testing"
+)
+
+func TestConn(t *testing.T) {
+	message := "Hi there!\n"
+
+	go func() {
+		conn, err := net.Dial("tcp", "localhost:8080")
+		if err != nil {
+			t.Error(err)
+			return
+		}
+		defer conn.Close()
+
+		if _, err := fmt.Fprintf(conn, message); err != nil {
+			t.Error(err)
+			return
+		}
+	}()
+
+	l, err := net.Listen("tcp", "localhost:8080")
```

Comment:
does this open up a real socket? we want to avoid doing that on unit tests

Reply:
It does, I'm having great difficulty understanding how to roll a mocked version of the net websocket b/c there's so many complex dependencies and higher-order methods used in this module such as goroutines, goroutine counter, and buffer input/output scanner

Reply:
briefly looked into this and it might worth trying out using `.pipe` 

https://stackoverflow.com/questions/30688685/how-does-one-test-net-conn-in-unit-tests-in-golang

Reply:
Thank you, will look into this